inception-kb/src/main/java/de/tudarmstadt/ukp/inception/kb/KnowledgeBaseServiceImpl.java

```diff
@@ -686,7 +684,47 @@ public KBHandle update(KnowledgeBase kb, UpdateAction aAction)
 
         return resultList;
     }
+    
+    @Override
+    public List<KBHandle> listDomainProperties(KnowledgeBase kb, String aDomain,
+            boolean aIncludeInferred, boolean aAll)
+        throws QueryEvaluationException
+    {
+        List<KBHandle> resultList = read(kb, (conn) -> {
+            ValueFactory vf = conn.getValueFactory();
+            String QUERY = SPARQLQueryStore.PROPERTYLIST_DOMAIN_DEPENDENT;
+            TupleQuery tupleQuery = conn.prepareTupleQuery(QueryLanguage.SPARQL, QUERY);
+            tupleQuery.setBinding("aDomain", vf.createIRI(aDomain));
+            tupleQuery.setBinding("pLABEL", kb.getLabelIri());
+            tupleQuery.setIncludeInferred(aIncludeInferred);
 
+            return evaluateListQuery(tupleQuery, aAll);
+        });
+        
+        resultList.sort(Comparator.comparing(KBObject::getUiLabel));
```

Comment:
May be we could discuss the pros and cons for it. In general, right now we don't do the sorting in SPARQL

Reply:
@Rentier The sorting in SPARQL won't be easy because `getUiLabel` is not trivial. If there is a label, it returns that, otherwise, it returns a suffix of the IRI. Even if we implement this logic in SPARQL, the SPARQL engine won't have a chance of optimizing it using indexes. It may even be slower than sorting locally because the SPARQL engine may first apply the sorting and then apply limiting, meaning it would have to sort all data, not only the returned data. Although I generally agree that sorting should be one in queries and not afterwards, in this particular case, I think it is actually better to sort after the query completed.

Reply:
@naveen2507 could you please add comment to the sort statement explaining why we do the local sort instead of sorting in the query?